ts/spi/provider_frame.ts

```diff
@@ -344,6 +354,22 @@ export class ProviderFrame {
     return this.handleUnimplementedRequest(id, msg.RPC_MESSAGE_TYPES.proxy);
   }
 
+  private async handleCancelLastOperation(id: string) {
+    if (!this.requestInProgress || this.cancellable === null) {
+      // no request in progress
+      console.warn('No pending request to cancel.');
+      this.clientChannel.send(msg.errorMessage(
```

Comment:
If this goes here, then two messages are sent (the requestFailed and the cancelLastOperation). Is that intended?